### ISSUE TITLE
Bug: Invalid option '--env' in Flat Config

### ISSUE THREAD

**Issue by xianyue0224** [eslint/eslint#16788]
### Environment

Node version: v18.12.0
npm version: v9.2.0
Local ESLint version: v8.32.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.22000

### What parser are you using?

Default (Espree)

### What did you do?

I want to pass --env option to eslint via command line, but it prompts me this：Invalid option '--env' - perhaps you meant '-v'?
What？？
![image](https://user-images.githubusercontent.com/83801486/212560522-a8fd36e5-d6f7-42df-b8fa-7e9930d09a81.png)
That's not what the documentation says!
This is the command I typed：**pnpm eslint --env browser,node,es2022 .**

### What did you expect to happen?

nothing

### What actually happened?

nothing

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @xianyue0224, thanks for the issue!

Are you using the [current config file format](https://eslint.org/docs/latest/use/configure/configuration-files) (`.eslintrc.*`) or the [new config file format](https://eslint.org/docs/latest/use/configure/configuration-files-new) (`eslint.config.js`)?  Can you run `pnpm eslint --env browser,node,es2022 . --debug` and share the output?

**Comment by kecrily** [comment]
> I forgot to use the debug mode to debug

Can you provide the logs with the `--debug` flag?

> but I executed the **`pnpm eslint --help`** command to see that there is no `--env` option.

<img width="705" alt="image" src="https://user-images.githubusercontent.com/45708948/212721723-30e4dd0c-ef2e-48d4-a991-d5b8dc1d901d.png">

Can you double-check this? Because I didn't reproduce here

**Comment by xianyue0224** [comment]
It is 2:30 in the morning in the Chinese time zone, and I saw my email reminding me that you @kecrily  commented on my issues. I quickly turned on the computer to re-verify the problems I encountered. Now, I am very sure that when using the new format configuration file, an `env` option is missing.

I created a demo to demonstrate the problem。

The current directory structure, and the contents of the configuration file:
![image](https://user-images.githubusercontent.com/83801486/212745711-558154e1-f6b3-454d-84e3-69a63f34df54.png)
Execute the `pnpm eslint . `command, the output tells me that this configuration file is available:
![image](https://user-images.githubusercontent.com/83801486/212745904-a3a8cc43-0e1f-45f0-8730-12189436bb8d.png)
Now execute the `pnpm eslint . --env browser` command, and the output seen is as follows:
![image](https://user-images.githubusercontent.com/83801486/212746200-be62d2b8-0903-4767-a5ff-dee28b835e05.png)
Next, I execute the `pnpm eslint --help` command, and the output are as follows:
![image](https://user-images.githubusercontent.com/83801486/212746520-ec6736be-6a63-4c32-932b-e4831ebf81f6.png)
I looked carefully at each line from beginning to end and did not find `--env`。

Now change the configuration file back to json format：
![image](https://user-images.githubusercontent.com/83801486/212746707-3b0b2e04-0936-4af7-bb68-9a74047c98b4.png)
Execute the above three commands again, and their output is as follows:
![image](https://user-images.githubusercontent.com/83801486/212746841-5dbf1a73-d014-4b2f-9e97-5b67b17fa6d3.png)
![image](https://user-images.githubusercontent.com/83801486/212746930-7d15748b-7e2f-47d4-8155-0c359422f57e.png)
![image](https://user-images.githubusercontent.com/83801486/212747005-126a4385-7c69-4f00-8905-d4300862aa61.png)

**There is still a question, why the error messages finally obtained by the two configuration files are inconsistent?**

![image](https://user-images.githubusercontent.com/83801486/212747111-340a7d92-a802-4359-af97-60edd7df850b.png)

**Comment by mdjermanovic** [comment]
> The config file I use is `eslint.config.js`

This works as intended because the [new config system](https://eslint.org/docs/latest/use/configure/configuration-files-new) does not have the concept of environments, so with `eslint.config.js` config files it isn't possible to use the `env` key or the `--env` CLI option.

https://github.com/eslint/rfcs/blob/main/designs/2019-config-simplification/README.md#detailed-design

You can still specify globals using `languageOptions.globals` key in config files or using the `--globals` CLI option.

**Comment by OJFord** [comment]
@mdjermanovic Just for the avoidance of doubt: with the new config format, instead of  `browser: true` for example we should be using `document: true, console: true` and so on in `globals`?

**Comment by mdjermanovic** [comment]
> with the new config format, instead of `browser: true` for example we should be using `document: true, console: true` and so on in `globals`?

Yes, each global variable needs to be individually enabled in `languageOptions.globals`. For enabling all browser globals, you can use the [`globals`](https://github.com/sindresorhus/globals) package:

```js
import globals from "globals";

export default [
    {
        languageOptions: {
            globals: {
                ...globals.browser
            }
        }
    }
];

```

This enables all [these global variables](https://github.com/sindresorhus/globals/blob/64d801f42fd9a35617622b1b19010701352efe6d/globals.json#L363-L1089). Note that `true` means that the variable should be treated as writable, while `false` means that it should be treated as readonly (e.g., assigning to the variable would be reported by [`no-global-assign`](https://eslint.org/docs/latest/rules/no-global-assign)).

**Comment by OJFord** [comment]
Ah, great, thank you. 🙏🏻
